--- origsrc/mlt-0.5.4/configure	2010-04-19 23:32:40.000000000 -0500
+++ src/mlt-0.5.4/configure	2010-06-17 17:04:24.425856600 -0500
@@ -71,14 +71,18 @@ build_config()
 		[ "$cpu" != "" ] && echo "TARGETCPU=-mcpu=$cpu"
 		echo "OPTIMISATIONS=-O2 -pipe -fomit-frame-pointer"
 
-		echo "CFLAGS+=-Wall -fPIC -DPIC \$(TARGETARCH) \$(TARGETCPU) \$(OPTIMISATIONS) \$(MMX_FLAGS) \$(SSE_FLAGS) \$(SSE2_FLAGS) \$(DEBUG_FLAGS) \$(LARGE_FILE)"
-
 		case $targetos in
+		Cygwin)
+		[ "$(uname -m)" = "x86_64" ] && echo "ARCH_X86_64=1" && echo "CFLAGS+=-DARCH_X86_64"
+		echo "SHFLAGS=-shared"
+		pic_flags="-DPIC"
+		;;
 		Darwin)
 		sysctl -a hw | grep "x86_64: 1" > /dev/null && echo "ARCH_X86_64=1" && echo "CFLAGS+=-DARCH_X86_64"
 		echo "CFLAGS+=-D__DARWIN__ `sdl-config --cflags`"
 		echo "SHFLAGS=-dynamiclib"
 		echo "LDFLAGS+=`sdl-config --libs`"
+		pic_flags="-DPIC -fPIC"
 		;;
 		Linux)
 		[ "$(uname -m)" = "x86_64" ] && echo "ARCH_X86_64=1" && echo "CFLAGS+=-DARCH_X86_64"
@@ -88,6 +92,7 @@ build_config()
 		echo "LIBDL=-ldl"
 		echo "RDYNAMIC=-rdynamic"
 		echo "LDFLAGS+=-Wl,--no-undefined -Wl,--as-needed"
+		pic_flags="-DPIC -fPIC"
 		;;
 		FreeBSD)
 		[ "$(uname -m)" = "x86_64" ] && echo "ARCH_X86_64=1" && echo "CFLAGS+=-DARCH_X86_64"
@@ -96,10 +101,13 @@ build_config()
 		echo "SHFLAGS=-shared"
 		echo "RDYNAMIC=-rdynamic"
 		echo "LDFLAGS+=-Wl,--no-undefined -Wl,--as-needed"
+		pic_flags="-DPIC -fPIC"
 		;;
 		*)
+		pic_flags="-DPIC -fPIC"
 		;;
 		esac
+		echo "CFLAGS+=-Wall $pic_flags \$(TARGETARCH) \$(TARGETCPU) \$(OPTIMISATIONS) \$(MMX_FLAGS) \$(SSE_FLAGS) \$(SSE2_FLAGS) \$(DEBUG_FLAGS) \$(LARGE_FILE)"
 		echo "LIBSUF=$LIBSUF"
 	) > config.mak
 
@@ -168,6 +176,10 @@ export targetos=
 targetos=$(uname -s)
 # Chose appropriate suffix for libraries
 case $targetos in
+	CYGWIN*)
+	LIBSUF=".dll"
+	targetos="Cygwin"
+	;;
 	Darwin)
 	LIBSUF=".dylib"
 	;;
--- origsrc/mlt-0.5.4/src/framework/Makefile	2010-04-19 23:32:40.000000000 -0500
+++ src/mlt-0.5.4/src/framework/Makefile	2010-06-17 14:49:18.882246000 -0500
@@ -3,6 +3,11 @@ include ../../config.mak
 NAME = libmlt$(LIBSUF)
 TARGET = $(NAME).$(version)
 
+ifeq ($(targetos), Cygwin)
+IMPLIB = libmlt.dll.a
+TARGET = cygmlt-$(soversion).dll
+SHFLAGS += -Wl,--out-implib,$(IMPLIB)
+else
 ifneq ($(targetos), Darwin)
 NAME = libmlt$(LIBSUF)
 TARGET = $(NAME).$(version)
@@ -14,6 +19,7 @@ TARGET = libmlt.$(version)$(LIBSUF)
 SONAME = libmlt.$(soversion)$(LIBSUF)
 SHFLAGS += -install_name $(libdir)/$(SONAME) -current_version $(version) -compatibility_version $(soversion)
 endif
+endif
 
 OBJS = mlt_frame.o \
 	   mlt_geometry.o \
@@ -75,8 +81,10 @@ all:	$(TARGET)
 
 $(TARGET): $(OBJS)
 		$(CC) $(SHFLAGS) -o $@ $(OBJS) $(LDFLAGS)
+ifneq ($(targetos), Cygwin)
 		ln -sf $(TARGET) $(NAME)
 		ln -sf $(TARGET) $(SONAME)
+endif
 
 depend:	$(SRCS)
 	$(CC) -MM $(CFLAGS) $^ 1>.depend
@@ -89,9 +97,15 @@ clean:
 
 install:
 	install -d $(DESTDIR)$(libdir)
+ifeq ($(targetos), Cygwin)
+	install -d $(DESTDIR)$(bindir)
+	install -m 755 $(TARGET) $(DESTDIR)$(bindir)
+	install -m 755 $(IMPLIB) $(DESTDIR)$(libdir)
+else
 	install -m 755 $(TARGET) $(DESTDIR)$(libdir)
 	ln -sf $(TARGET) $(DESTDIR)$(libdir)/$(SONAME)
 	ln -sf $(TARGET) $(DESTDIR)$(libdir)/$(NAME)
+endif
 	install -d "$(DESTDIR)$(prefix)/include/mlt/framework"
 	install -m 644 $(INCS) "$(DESTDIR)$(prefix)/include/mlt/framework"
 	install -d "$(DESTDIR)$(datadir)/mlt"
--- origsrc/mlt-0.5.4/src/mlt++/Makefile	2010-04-19 23:32:40.000000000 -0500
+++ src/mlt-0.5.4/src/mlt++/Makefile	2010-06-17 14:50:14.980454700 -0500
@@ -2,6 +2,11 @@ include ../../config.mak
 include config.mak
 INSTALL = install
 
+ifeq ($(targetos), Cygwin)
+IMPLIB = libmlt++.dll.a
+TARGET = cygmlt++-$(soversion).dll
+LIBFLAGS += -Wl,--out-implib,$(IMPLIB)
+else
 ifneq ($(targetos), Darwin)
 NAME = libmlt++$(LIBSUF)
 TARGET = $(NAME).$(version)
@@ -13,6 +18,7 @@ TARGET = libmlt++.$(version)$(LIBSUF)
 SONAME = libmlt++.$(soversion)$(LIBSUF)
 LIBFLAGS += -install_name $(libdir)/$(SONAME) -current_version $(version) -compatibility_version $(soversion)
 endif
+endif
 
 CXXFLAGS += -I.. $(RDYNAMIC) -DVERSION=\"$(version)\"
 
@@ -48,8 +54,10 @@ all:		$(TARGET)
 
 $(TARGET):	$(OBJS)
 	$(CXX) $(LIBFLAGS) -o $@ $(OBJS) $(LDFLAGS)
+ifneq ($(targetos), Cygwin)
 	ln -sf $(TARGET) $(NAME)
 	ln -sf $(TARGET) $(SONAME)
+endif
 
 depend:	$(SRCS)
 	$(CXX) -MM $(CXXFLAGS) $^ 1>.depend
@@ -61,9 +69,15 @@ distclean:	clean
 
 install:
 	$(INSTALL) -d "$(DESTDIR)$(libdir)"
+ifeq ($(targetos), Cygwin)
+	install -d $(DESTDIR)$(bindir)
+	install -m 755 $(TARGET) $(DESTDIR)$(bindir)
+	install -m 755 $(IMPLIB) $(DESTDIR)$(libdir)
+else
 	$(INSTALL) -m 755 $(TARGET) $(DESTDIR)$(libdir)
 	ln -sf $(TARGET) $(DESTDIR)$(libdir)/$(NAME)
 	ln -sf $(TARGET) $(DESTDIR)$(libdir)/$(SONAME)
+endif
 	$(INSTALL) -d "$(DESTDIR)$(prefix)/include/mlt++"
 	$(INSTALL) -m 644 $(HEADERS) "$(DESTDIR)$(prefix)/include/mlt++"
 
--- origsrc/mlt-0.5.4/src/mlt++/configure	2010-04-19 23:32:40.000000000 -0500
+++ src/mlt-0.5.4/src/mlt++/configure	2010-06-17 14:51:51.471973600 -0500
@@ -6,6 +6,11 @@ WARNINGS="-W -Wwrite-strings -Wcast-qual
 
 targetos=$(uname -s)
 case $targetos in 
+	CYGWIN*)
+		echo LIBSUF=.dll
+		echo "CXXFLAGS+=-Wall $WARNINGS -DPIC"
+		echo "LIBFLAGS=-shared"
+		;;
 	Darwin)
 		echo LIBSUF=.dylib
 		echo "CXXFLAGS+=-D__DARWIN__ -Wall -fPIC"
--- origsrc/mlt-0.5.4/src/modules/kino/Makefile	2010-04-19 23:32:40.000000000 -0500
+++ src/mlt-0.5.4/src/modules/kino/Makefile	2010-06-17 17:02:28.365218300 -0500
@@ -6,7 +6,7 @@ LDFLAGS += -L../../framework -lmlt -lpth
 include ../../../config.mak
 include config.mak
 
-TARGET = ../libmltkino.so
+TARGET = ../libmltkino$(LIBSUF)
 
 OBJS = factory.o producer_kino.o
 CPPOBJS = kino_wrapper.o avi.o error.o filehandler.o riff.o
--- origsrc/mlt-0.5.4/src/modules/motion_est/Makefile	2010-04-19 23:32:40.000000000 -0500
+++ src/mlt-0.5.4/src/modules/motion_est/Makefile	2010-06-17 17:02:46.654264400 -0500
@@ -4,7 +4,7 @@ LDFLAGS += -L../../framework -lmlt -lm
 
 include ../../../config.mak
 
-TARGET = ../libmltmotion_est.so
+TARGET = ../libmltmotion_est$(LIBSUF)
 
 OBJS = factory.o \
 	   filter_motion_est.o \
--- origsrc/mlt-0.5.4/src/modules/qimage/configure	2010-04-19 23:32:40.000000000 -0500
+++ src/mlt-0.5.4/src/modules/qimage/configure	2010-06-17 16:57:13.315198500 -0500
@@ -16,7 +16,7 @@ EOF
 else
 	targetos=$(uname -s)
 	case $targetos in
-	MINGW32*)
+	CYGWIN*|MINGW32*)
 		export LIBSUF=.dll
 		;;
 	Darwin)
