--- origsrc/mlt-0.8.8/src/modules/videostab/stab/estimate.c	2013-01-20 23:57:25.000000000 -0600
+++ src/mlt-0.8.8/src/modules/videostab/stab/estimate.c	2013-09-24 23:56:29.479198600 -0500
@@ -18,7 +18,7 @@
 #include <stdlib.h>
 #include <math.h>
 #include <string.h>
-#if !defined(__DARWIN__) && !defined(__FreeBSD__) && !defined(WIN32) && !defined(__NetBSD__)
+#if !defined(__DARWIN__) && !defined(__FreeBSD__) && !defined(WIN32) && !defined(__NetBSD__) && !defined(__CYGWIN__)
 #include <values.h>
 #endif
 
--- origsrc/mlt-0.8.8/src/modules/videostab/stab/klt/base.h	2013-01-20 23:57:25.000000000 -0600
+++ src/mlt-0.8.8/src/modules/videostab/stab/klt/base.h	2013-09-24 23:56:51.954484100 -0500
@@ -13,6 +13,7 @@
 #define schar signed char
 #endif
 
+#ifndef __CYGWIN__
 #ifndef uint
 #define uint unsigned int
 #endif
@@ -24,6 +25,7 @@
 #ifndef ulong
 #define ulong unsigned long
 #endif
+#endif /* !__CYGWIN__ */
 
 #ifndef max
 #define max(a,b)	((a) > (b) ? (a) : (b))
