--- origsrc/mlt-0.8.8/src/modules/gtk2/producer_pango.c	2013-01-20 23:57:25.000000000 -0600
+++ src/mlt-0.8.8/src/modules/gtk2/producer_pango.c	2014-04-07 12:32:17.088484900 -0500
@@ -25,7 +25,8 @@
 #include <string.h>
 #include <gdk-pixbuf/gdk-pixbuf.h>
 #include <pango/pangoft2.h>
-#include <freetype/freetype.h>
+#include <ft2build.h>
+#include FT_FREETYPE_H
 #include <iconv.h>
 #include <pthread.h>
 #include <ctype.h>
