--- origsrc/mlt-0.9.2/configure	2014-06-29 15:23:17.000000000 -0500
+++ src/mlt-0.9.2/configure	2014-08-26 22:29:01.818610900 -0500
@@ -130,6 +130,12 @@ build_config()
 		echo "RDYNAMIC="
 		echo "LDFLAGS+=-Wl,--no-undefined -Wl,--as-needed"
 		;;		
+		Cygwin)
+		[ "$(uname -m)" = "x86_64" ] && echo "ARCH_X86_64=1" && echo "CFLAGS+=-DARCH_X86_64"
+		[ "$optimisations" = "true" ] &&
+			echo "OPTIMISATIONS+=-ffast-math"
+		echo "SHFLAGS=-shared"
+		;;
 		*)
 		;;
 		esac
@@ -266,6 +272,10 @@ case $targetos in
 	targetos="MinGW"
 	LIBSUF=".dll"
 	;;
+	CYGWIN_NT-*)
+	targetos="Cygwin"
+	LIBSUF=".dll"
+	;;
 	*)
 	LIBSUF=".so"
 	;;
--- origsrc/mlt-0.9.2/src/framework/Makefile	2014-06-29 15:23:17.000000000 -0500
+++ src/mlt-0.9.2/src/framework/Makefile	2014-08-26 22:29:01.825111700 -0500
@@ -13,6 +13,10 @@ else ifeq ($(targetos), MinGW)
 NAME = libmlt$(LIBSUF)
 TARGET = libmlt-$(soversion)$(LIBSUF)
 SHFLAGS += -Wl,--output-def,libmlt.def
+else ifeq ($(targetos), Cygwin)
+NAME = cygmlt$(LIBSUF)
+TARGET = cygmlt-$(soversion)$(LIBSUF)
+SHFLAGS += -Wl,--out-implib,libmlt.dll.a
 else
 NAME = libmlt$(LIBSUF)
 TARGET = $(NAME).$(version)
@@ -95,7 +99,7 @@ all:	$(TARGET)
 $(TARGET): $(OBJS)
 		$(CC) $(SHFLAGS) -o $@ $(OBJS) $(LDFLAGS)
 		ln -sf $(TARGET) $(NAME)
-		if [ "$(targetos)" != "MinGW" ]; then \
+		if [ "$(targetos)" != "MinGW" -a "$(targetos)" != "Cygwin" ]; then \
 			ln -sf $(TARGET) $(SONAME) ; \
 		fi
 
@@ -114,6 +118,10 @@ install:
 		install -m 755 $(TARGET) $(DESTDIR)$(prefix) ; \
 		install -m 755 $(TARGET) $(DESTDIR)$(libdir)/libmlt.dll ; \
 		install -m 644 libmlt.def $(DESTDIR)$(libdir) ; \
+	elif [ "$(targetos)" = "Cygwin" ]; then \
+		install -d $(DESTDIR)$(bindir) ; \
+		install -m 755 $(TARGET) $(DESTDIR)$(bindir) ; \
+		install -m 644 libmlt.dll.a $(DESTDIR)$(libdir) ; \
 	else \
 		install -m 755 $(TARGET) $(DESTDIR)$(libdir) ; \
 		ln -sf $(TARGET) $(DESTDIR)$(libdir)/$(SONAME) ; \
--- origsrc/mlt-0.9.2/src/mlt++/Makefile	2014-06-29 15:23:17.000000000 -0500
+++ src/mlt-0.9.2/src/mlt++/Makefile	2014-08-26 22:29:01.830612400 -0500
@@ -12,6 +12,11 @@ NAME = libmlt++$(LIBSUF)
 TARGET = libmlt++-$(soversion)$(LIBSUF)
 CXXFLAGS += -DMLTPP_EXPORTS
 LIBFLAGS += -Wl,--output-def,libmlt++.def
+else ifeq ($(targetos), Cygwin)
+NAME = cygmlt++$(LIBSUF)
+TARGET = cygmlt++-$(soversion)$(LIBSUF)
+CXXFLAGS += -DMLTPP_EXPORTS
+LIBFLAGS += -Wl,--out-implib,libmlt++.dll.a
 else
 NAME = libmlt++$(LIBSUF)
 TARGET = $(NAME).$(version)
@@ -57,7 +62,7 @@ all:		$(TARGET)
 $(TARGET):	$(OBJS)
 	$(CXX) $(LIBFLAGS) -o $@ $(OBJS) $(LDFLAGS)
 	ln -sf $(TARGET) $(NAME)
-	if [ "$(targetos)" != "MinGW" ]; then \
+	if [ "$(targetos)" != "MinGW" -a "$(targetos)" != "Cygwin" ]; then \
 		ln -sf $(TARGET) $(SONAME) ; \
 	fi
 
@@ -75,6 +80,10 @@ install:
 		$(INSTALL) -m 755 $(TARGET) $(DESTDIR)$(prefix) ; \
 		$(INSTALL) -m 755 $(TARGET) $(DESTDIR)$(libdir)/libmlt++.dll ; \
 		$(INSTALL) -m 644 libmlt++.def $(DESTDIR)$(libdir) ; \
+	elif [ "$(targetos)" = "Cygwin" ]; then \
+		$(INSTALL) -d "$(DESTDIR)$(bindir)" ; \
+		$(INSTALL) -m 755 $(TARGET) $(DESTDIR)$(bindir) ; \
+		$(INSTALL) -m 644 libmlt++.dll.a $(DESTDIR)$(libdir) ; \
 	else \
 		$(INSTALL) -m 755 $(TARGET) $(DESTDIR)$(libdir) ; \
 		ln -sf $(TARGET) $(DESTDIR)$(libdir)/$(SONAME) ; \
--- origsrc/mlt-0.9.2/src/mlt++/configure	2014-06-29 15:23:17.000000000 -0500
+++ src/mlt-0.9.2/src/mlt++/configure	2014-08-26 22:29:01.836113100 -0500
@@ -15,7 +15,7 @@ case $targetos in
 		echo "CXXFLAGS+=-Wall $WARNINGS -fPIC -DPIC"
 		echo "LIBFLAGS=-shared"
 		;;
-	MinGW)
+	MinGW|Cygwin)
 		echo LIBSUF=.dll
 		echo "CXXFLAGS+=-Wall $WARNINGS -DPIC"
 		echo "LIBFLAGS=-Wl,-enable-auto-import -shared"
--- origsrc/mlt-0.9.2/src/modules/kino/Makefile	2014-06-29 15:23:17.000000000 -0500
+++ src/mlt-0.9.2/src/modules/kino/Makefile	2014-08-26 22:29:01.841613800 -0500
@@ -6,7 +6,7 @@ CXXFLAGS := $(CFLAGS) -Wno-deprecated $(
 
 LDFLAGS := -L../../framework -lmlt -lpthread $(LDFLAGS)
 
-TARGET = ../libmltkino.so
+TARGET = ../libmltkino$(LIBSUF)
 
 OBJS = factory.o producer_kino.o
 CPPOBJS = kino_wrapper.o avi.o error.o filehandler.o riff.o
