--- origsrc/mlt-0.9.2/src/swig/csharp/build	2014-06-29 15:23:17.000000000 -0500
+++ src/mlt-0.9.2/src/swig/csharp/build	2014-08-26 22:31:15.378570800 -0500
@@ -14,13 +14,13 @@ then
 
 	# Invoke swig
 	mkdir src_swig
-	swig -c++ -I../../mlt++ -I../.. -csharp -dllimport libmltsharp -outdir src_swig -namespace Mlt mlt.i || exit $?
+	swig -c++ -I../../mlt++ -I../.. -csharp -dllimport mltsharp -outdir src_swig -namespace Mlt mlt.i || exit $?
 
 	# Compile the wrapper
 	g++ -fPIC -D_GNU_SOURCE ${CXXFLAGS} -c -rdynamic -pthread -I../.. mlt_wrap.cxx || exit $?
 	
 	# Create the module
-	g++ ${CXXFLAGS} -shared mlt_wrap.o -L../../mlt++ -lmlt++ -o libmltsharp.so || exit $?
+	g++ ${CXXFLAGS} -shared mlt_wrap.o -L../../mlt++ -lmlt++ -L../../framework -lmlt -o cygmltsharp.dll || exit $?
 
 	# Compile the library assembly
 	mcs -out:mlt-sharp.dll -target:library src_swig/*.cs
--- origsrc/mlt-0.9.2/src/swig/java/build	2014-06-29 15:23:17.000000000 -0500
+++ src/mlt-0.9.2/src/swig/java/build	2014-08-26 22:31:15.382571300 -0500
@@ -27,13 +27,13 @@ then
 	g++ -fPIC -D_GNU_SOURCE ${CXXFLAGS} -c -rdynamic -pthread -I../.. mlt_wrap.cxx $JAVA_INCLUDE || exit $?
 	
 	# Create the module
-	g++ ${CXXFLAGS} -shared mlt_wrap.o -L../../mlt++ -lmlt++ -o libmlt_java.so || exit $?
+	g++ ${CXXFLAGS} -shared mlt_wrap.o -L../../mlt++ -lmlt++ -L../../framework -lmlt -o cygmlt_java.dll || exit $?
 
 	# Compile the test
 	javac `find src_swig -name '*.java'` || exit $?
 
 	export CLASSPATH=`pwd`/src_swig
-	javac Play.java
+	javac -classpath $CLASSPATH Play.java
 else
 	echo "Java command not found"
 	exit 1
--- origsrc/mlt-0.9.2/src/swig/lua/build	2014-06-29 15:23:17.000000000 -0500
+++ src/mlt-0.9.2/src/swig/lua/build	2014-08-26 22:31:15.387071900 -0500
@@ -19,7 +19,7 @@ then
 	g++ -fPIC -DPIC -D_GNU_SOURCE ${CXXFLAGS} -c -rdynamic -pthread -I../.. mlt_wrap.cxx || exit $?
 
 	# Create the module
-	g++ ${CXXFLAGS} -shared mlt_wrap.o -L../../mlt++ -lmlt++ -o mlt.so || exit $?
+	g++ ${CXXFLAGS} -shared mlt_wrap.o -L../../mlt++ -lmlt++ -L../../framework -lmlt -llua -o mlt.dll || exit $?
 else
 	echo Lua not installed.
 	exit 1
--- origsrc/mlt-0.9.2/src/swig/perl/Makefile.PL	2014-06-29 15:23:17.000000000 -0500
+++ src/mlt-0.9.2/src/swig/perl/Makefile.PL	2014-08-26 22:31:15.390572300 -0500
@@ -9,7 +9,7 @@ WriteMakefile(
 	'NAME'    => 'mlt',
 	'CC'      => '${CXX} -fPIC ${CXXFLAGS} -I../..',
 	'OPTIMIZE' => '-O2 -g -pipe -Wp,-D_FORTIFY_SOURCE=2 -fexceptions',
-	'LIBS'    => ['-L../../mlt++ -lmlt++'],
+	'LIBS'    => ['-L../../mlt++ -lmlt++ -L../../framework -lmlt'],
 	'OBJECT'  => 'mlt_wrap.o',
 	'DESTDIR' => $ENV{'DESTDIR'},
 );
--- origsrc/mlt-0.9.2/src/swig/php/build	2014-06-29 15:23:17.000000000 -0500
+++ src/mlt-0.9.2/src/swig/php/build	2014-08-26 22:31:15.394572900 -0500
@@ -10,4 +10,4 @@ fi
 ln -sf ../mlt.i
 swig -c++ -I../../mlt++ -I../.. -php5 -noproxy mlt.i
 g++ -fPIC -DPIC -D_GNU_SOURCE ${CXXFLAGS} -c -rdynamic -pthread -I../.. `php-config --includes` mlt_wrap.cpp
-g++ ${CXXFLAGS} -shared mlt_wrap.o -L../../mlt++ -lmlt++ -o mlt.so || exit $?
+g++ ${CXXFLAGS} -shared mlt_wrap.o -L../../mlt++ -lmlt++ -L../../framework -lmlt -lphp5 -o mlt.dll || exit $?
--- origsrc/mlt-0.9.2/src/swig/php/play.php	2014-06-29 15:23:17.000000000 -0500
+++ src/mlt-0.9.2/src/swig/php/play.php	2014-08-26 22:31:15.398073300 -0500
@@ -1,6 +1,6 @@
 <?php
 $filename = $argv[1];
-dl("mlt.so");
+dl("mlt." . PHP_SHLIB_SUFFIX);
 mlt_factory_init(NULL);
 $profile = new_profile("dv_ntsc");
 $p = new_producer( $profile, $filename );
--- origsrc/mlt-0.9.2/src/swig/python/build	2014-06-29 15:23:17.000000000 -0500
+++ src/mlt-0.9.2/src/swig/python/build	2014-08-26 22:31:15.401573700 -0500
@@ -24,7 +24,7 @@ then
 	g++ -fPIC -D_GNU_SOURCE ${CXXFLAGS} -c -I../.. -I$PYTHON_INCLUDE mlt_wrap.cxx || exit $?
 
 	# Create the module
-	g++ ${CXXFLAGS} -shared mlt_wrap.o -L../../mlt++ -lmlt++ -L../../framework -lmlt $(python-config --ldflags) -o _mlt.so || exit $?
+	g++ ${CXXFLAGS} -shared mlt_wrap.o -L../../mlt++ -lmlt++ -L../../framework -lmlt $(python-config --ldflags) -o _mlt.dll || exit $?
 else
 	echo Python not installed.
 	exit 1
--- origsrc/mlt-0.9.2/src/swig/ruby/build	2014-06-29 15:23:17.000000000 -0500
+++ src/mlt-0.9.2/src/swig/ruby/build	2014-08-26 22:31:15.404574100 -0500
@@ -7,7 +7,7 @@
 end
 system( "ln -sf ../mlt.i" )
 system( "swig -c++ -ruby -I../../mlt++ -I../.. mlt.i" )
-$CFLAGS += " -I../.. " + (ENV.has_key?('CXXFLAGS')? ENV['CXXFLAGS'] : '')
-$LDFLAGS += " -L../../mlt++ -lmlt++"
+$INCFLAGS += " -I../.. " + (ENV.has_key?('CXXFLAGS')? ENV['CXXFLAGS'] : '')
+$LDFLAGS += " -L../../mlt++ -lmlt++ -L../../framework -lmlt -lstdc++"
 create_makefile('mlt')
 system( "make" )
--- origsrc/mlt-0.9.2/src/swig/tcl/build	2014-06-29 15:23:17.000000000 -0500
+++ src/mlt-0.9.2/src/swig/tcl/build	2014-08-26 22:31:15.407074400 -0500
@@ -16,10 +16,10 @@ then
 	swig -c++ -I../../mlt++ -I../.. -tcl mlt.i || exit 1
 
 	# Compile the wrapper
-	g++ -fPIC -D_GNU_SOURCE ${CXXFLAGS} -c -rdynamic -pthread -I../.. mlt_wrap.cxx || exit 1
+	g++ -fPIC -DUSE_TCL_STUBS -D_GNU_SOURCE ${CXXFLAGS} -c -rdynamic -pthread -I../.. mlt_wrap.cxx || exit 1
 
 	# Create the module
-	g++ ${CXXFLAGS} -shared mlt_wrap.o -L../../mlt++ -lmlt++ -o mlt.so || exit 1
+	g++ ${CXXFLAGS} -shared mlt_wrap.o -L../../mlt++ -lmlt++ -L../../framework -lmlt -ltclstub -o mlt.dll || exit 1
 else
 	echo "Unable to locate tclsh."
 	exit 1
--- origsrc/mlt-0.9.2/src/swig/tcl/play.tcl	2014-06-29 15:23:17.000000000 -0500
+++ src/mlt-0.9.2/src/swig/tcl/play.tcl	2014-08-26 22:31:15.409074700 -0500
@@ -1,6 +1,6 @@
 #!/usr/bin/env tclsh
 
-load mlt.so
+load mlt
 Factory_init
 set profile [Profile]
 set arg1 [lindex $argv 0]
